lib/classes/ship.dart

```diff
@@ -0,0 +1,87 @@
+import 'dart:math';
+
+import 'package:phone_factory/classes/bay_item.dart';
+
+class Ship {
+  static final Ship _instance = Ship._internal();
```

Comment:
no need singleton pattern for now. You need to create your Ship one time in Game class and path data to other classes using method arguments and that's constructors parameters